src/telemetry/types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Result type constants for telemetry events
+ */
+export const TELEMETRY_RESULT = {
+    SUCCESS: "success" as const,
+    FAILURE: "failure" as const,
+};
+
+export type TelemetryResult = (typeof TELEMETRY_RESULT)[keyof typeof TELEMETRY_RESULT];
```

Comment:
```suggestion
export type TelemetryResult = "success" | "failure";
```
I don't think we benefit much from this enum-like constant, I'd just do a string-based type definition and use string directly; enums are weird in JS from my experience and we get all benefits from auto-completion for union string types anyways

Reply:
updated!